caddyhttp/httpserver/server.go

```diff
@@ -406,3 +488,15 @@ func WriteTextResponse(w http.ResponseWriter, status int, body string) {
 	w.WriteHeader(status)
 	w.Write([]byte(body))
 }
+
+// IsCommonError checks if the given error is an error that could pop up
+// when handling request. Currently only checks for MaxBytesReader error.
+// Returns a boolean - whether the given error is a common error, and an int -
+// the advisory HTTP status code to return to the user (or -1 if the given error
+// is not a common error)
+func IsCommonError(err error) (bool, int) {
```

Comment:
I still don't quite feel good about this function. It's only used by MaxBytesReader, I think error handling should not be here. It should probably remain in the handlers that use the MaxBytesReader.